R/formula.R

```diff
@@ -0,0 +1,129 @@
+new_googlesheets4_formula <- function(x = character()) {
+  vec_assert(x, character())
+  new_vctr(x, class = "googlesheets4_formula")
+}
+
+#' `googlesheets4_formula` class
+#'
+#' In order to write a formula into Google Sheets, you need to store it as an
+#' object of class `googlesheets4_formula`. This is how we distinguish a
+#' "regular" character string from a string that should be interpreted as a
+#' formula. `googlesheets4_formula` is an S3 class implemented using the [vctrs
+#' package](https://vctrs.r-lib.org/articles/s3-vector.html).
+#'
+#' @param x Character.
+#'
+#' @return An S3 vector of class `googlesheets4_formula`.
+#' @export
+#' @family write functions
+#'
+#' @examples
+#' if (sheets_has_token()) {
+#' dat <- data.frame(X = c(1, 5, 3, 2, 4, 6))
+#'
+#' ss <- sheets_write(dat)
+#'
+#' summaries <- tibble::tribble(
+#'        ~ desc, ~ summaries,
+#'         "max", "=max(A:A)",
+#'         "sum", "=sum(A:A)",
+#'         "min", "=min(A:A)",
+#'   "sparkline", "=SPARKLINE(A:A, {\"color\", \"blue\"})"
```

Comment:
Maybe use `'` to avoid so much escaping?

Reply:
I could use single quotes on the R side? Believe it or not, you can't actually use `'` to quote in these formulas.

Reply:
I think I'll leave it. The requirement to use double quotes is so weird that it might actually be a public service to draw the user's attention to it.